metaflow/plugins/pypi/pip.py

```diff
@@ -156,29 +154,30 @@ def metadata(self, id_, packages, python, platform):
         with open(metadata_file, "r") as file:
             return json.loads(file.read())
 
-    def extra_index_urls(self, prefix):
-        # get extra index urls from Pip conf
+    def indices(self, prefix):
+        indices = []
         extra_indices = []
-        for key in [":env:.extra-index-url", "global.extra-index-url"]:
-            try:
-                extras = self._call(prefix, args=["config", "get", key], isolated=False)
-                extra_indices.extend(extras.split(" "))
-            except Exception:
-                # Pip will throw an error when trying to get a config key that does
-                # not exist
-                pass
-        return extra_indices
-
-    def index_url(self, prefix):
-        # get custom index url from Pip conf
         try:
-            return self._call(
-                prefix, args=["config", "get", "global.index-url"], isolated=False
-            )
+            config = self._call(prefix, args=["config", "list"], isolated=False)
+            for line in config.splitlines():
+                key, value = line.split("=", 1)
+                _, key = key.split(".")
+                value = value.strip("'\"")
+                if key == "index-url":
+                    indices.append(value)
+                elif key == "extra-index-url":
+                    extra_indices.append(value)
```

Comment:
Am fixing this a bit so may as well take the fix: https://github.com/Netflix/metaflow-nflx-extensions/pull/29